src/main/java/org/apache/commons/compress/archivers/sevenz/LZMADecoder.java

```diff
@@ -34,11 +35,13 @@
 
     @Override
     InputStream decode(final String archiveName, final InputStream in, final long uncompressedLength,
-            final Coder coder, final byte[] password) throws IOException {
+            final Coder coder, final byte[] password, int maxMemoryLimitInKb) throws IOException {
         final byte propsByte = coder.properties[0];
         final int dictSize = getDictionarySize(coder);
         if (dictSize > LZMAInputStream.DICT_SIZE_MAX) {
             throw new IOException("Dictionary larger than 4GiB maximum size used in " + archiveName);
+        } else if ((dictSize / 1024) > maxMemoryLimitInKb) {
```

Comment:
I think you should call `LZMAInputStream.getMemoryUsage` rather than evaluate the dictionary size only. Too bad the constructor we use doesn't support the `memoryLimit` parameter other constructors allow.

Reply:
Didn't know about that. I changed the code according to your comment. Do